scripts/release.sh

```diff
@@ -43,31 +41,58 @@ fi
 develop_branch="develop"
 main_branch="master"
 
-# Stash any local changes to avoid errors during checkout
-git status
-prompt_user "Would you like to stash any local changes"
-if [[ $? == 1 ]]; then
-    # Stash local changes to prevent issues with checkout
-    git stash
-    echo "use \"git stash pop\" when this script is complete to restore your changes"
+# Checkout develop - so we can update versions.
+# We need to checkout the branch before we start modifying files.
+current_branch=$(git branch --show-current)
+if [ $current_branch == $develop_branch ]; then
+    echo
+    echo "$develop_branch is already checked out."
 else
-    # Dump local changes to prevent issues with checkout
-    git reset --hard
+    echo
+    echo "Checkout of $develop_branch is required for some steps"
+    prompt_user "Would you like to automatically checkout $develop_branch"
+    if [[ $? == 1 ]]; then
+
+        # Stash any local changes to avoid errors during checkout
+        changes=$(git diff-files)
+        if [ ! -z "$changes" ]; then   
+            echo "There are unstaged changes in these files"
+            echo $changes
+            prompt_user "Would you like to stash these local changes before checkout of $develop_branch"
+            if [[ $? == 1 ]]; then
+                # Stash local changes to prevent issues with checkout
+                git stash
+                echo "use \"git stash pop\" when this script is complete to restore your changes"
+            else
+                # Dump local changes to prevent issues with checkout
+                git reset --hard
+            fi
+        fi
+        
+        # Do a pull to make sure we are up to date.
+        git pull --ff
```

Comment:
It would probably be better to detect any kind of `git pull` failure, so I have added detection for that.  `git-pull` already reports out the failure, so I do not need an extra `echo` for that.